src/pretix/base/models/orders.py

```diff
@@ -229,7 +229,14 @@ def can_modify_answers(self) -> bool:
 
     @property
     def can_user_cancel(self) -> bool:
-        return self.event.settings.cancel_allow_user
+        positions = self.positions.all().prefetch_related('item')
+        cancelable = True
+        cancelable_products = []
+        for i, op in enumerate(positions):
+            cancelable_products.append(op.item.allow_cancel)
+        if False in cancelable_products:
```

Comment:
I think this should be `cancelable = all([op.item.allow_cancel for op in positions])` without the `not`